modules/build/src/main/scala/scala/build/preprocessing/mdsandbox/MarkdownSnippet.scala

```diff
@@ -0,0 +1,121 @@
+package scala.build.preprocessing.mdsandbox
+
+import scala.collection.mutable
+
+object MarkdownSnippet {
+  private val allowIndentedFence: Boolean = false
+
+  case class Fence(
+    info: Seq[String],
+    body: String,
+    startLine: Int,  // start of fenced body EXCLUDING backticks
+    endLine: Int     // same as above
+  ) {
+    override def toString(): String = s"Fence[$info, lines $startLine-$endLine]{${body.replace("\n", "\\n")}}"
+
+    /** 
+      * @return `true` if this snippet should be ignored, `false` otherwise 
+      */
+    def shouldIgnore: Boolean = info(0) != "scala" || info.contains("ignore")
+
+    /**
+      * @return `true` if this snippet should have its scope reset, `false` otherwise
+      */
+    def resetScope: Boolean = info.contains("reset")
```

Comment:
I don't particularly like using `reset` as the keyword for separating scopes. 
something like `--fresh-scope`, `--markdown-flush`, `--fresh-wrapper` (or anything less generalizing, really) would be better, imo.